python/ray/experimental/channel/shared_memory_channel.py

```diff
@@ -420,6 +420,9 @@ def write(self, value: Any):
             self._num_readers,
         )
 
+    def read(self) -> Any:
+        return self.begin_read()
+
     def begin_read(self) -> Any:
```

Comment:
Yes, good idea, let's do that.

It would be great if we could remove the `begin_read()` and `end_read()` calls completely but if that's too complicated, we can leave it for follow-up.